internal/testutil/query.go

```diff
@@ -20,25 +20,43 @@ func Query(query string, res any, opts *QueryOptions) error {
 }
 
 func QueryWithEngineConfig(query string, res any, opts *QueryOptions, cfg *engine.Config) error {
```

Comment:
nit: maybe this could become `QueryWithClientOpts(..., opts ...dagger.ClientOpt)`?

Reply:
Done!